sdk/python/feast/infra/offline_stores/remote.py

```diff
@@ -0,0 +1,413 @@
+        end_date: datetime,
+    ) -> RetrievalJob:
+        assert isinstance(config.offline_store, RemoteOfflineStoreConfig)
+
+        # Initialize the client connection
+        client = RemoteOfflineStore.init_client(config)
+
+        api_parameters = {
+            "data_source_name": data_source.name,
+            "join_key_columns": join_key_columns,
+            "feature_name_columns": feature_name_columns,
+            "timestamp_field": timestamp_field,
+            "created_timestamp_column": created_timestamp_column,
+            "start_date": start_date.isoformat(),
+            "end_date": end_date.isoformat(),
+        }
+
+        return RemoteRetrievalJob(
+            client=client,
+            api=OfflineStore.pull_latest_from_table_or_query.__name__,
+            api_parameters=api_parameters,
+        )
+
+    @staticmethod
+    def write_logged_features(
+        config: RepoConfig,
+        data: Union[pyarrow.Table, Path],
+        source: LoggingSource,
+        logging_config: LoggingConfig,
+        registry: BaseRegistry,
+    ):
+        assert isinstance(config.offline_store, RemoteOfflineStoreConfig)
+        assert isinstance(source, FeatureServiceLoggingSource)
+
+        if isinstance(data, Path):
+            data = pyarrow.parquet.read_table(data, use_threads=False, pre_buffer=False)
+
+        # Initialize the client connection
+        client = RemoteOfflineStore.init_client(config)
+
+        api_parameters = {
+            "feature_service_name": source._feature_service.name,
+        }
+
+        api_name = OfflineStore.write_logged_features.__name__
+
+        command_descriptor = _call_put(
+            api=api_name,
+            api_parameters=api_parameters,
+            client=client,
+            table=data,
+            entity_df=None,
+        )
+        bytes = command_descriptor.serialize()
+
+        client.do_action(pa.flight.Action(api_name, bytes))
+
+    @staticmethod
+    def offline_write_batch(
+        config: RepoConfig,
+        feature_view: FeatureView,
+        table: pyarrow.Table,
+        progress: Optional[Callable[[int], Any]],
+    ):
+        assert isinstance(config.offline_store, RemoteOfflineStoreConfig)
+
+        # Initialize the client connection
+        client = RemoteOfflineStore.init_client(config)
+
+        feature_view_names = [feature_view.name]
+        name_aliases = [feature_view.projection.name_alias]
+
+        api_parameters = {
+            "feature_view_names": feature_view_names,
+            "progress": progress,
+            "name_aliases": name_aliases,
+        }
+
+        api_name = OfflineStore.offline_write_batch.__name__
+        command_descriptor = _call_put(
+            api=api_name,
+            api_parameters=api_parameters,
+            client=client,
+            table=table,
+            entity_df=None,
+        )
+        bytes = command_descriptor.serialize()
+
+        client.do_action(pa.flight.Action(api_name, bytes))
```

Comment:
does this need to happen with two commands? I think a single put should be enough. I get that this way it's more similar to other actions, but we'd be essentially forcing the whole dataset to be kept in memory on server until do_action is called.

Reply:
You might have a point but in case of `persist()` I need 2 parameters one is the retrieve method to call on the server and the second to know that I need to call persist, do you have an elegant solution without adding specific code to handle this flow?

Reply:
@tmihalac just to confirm is this resolved ?

Reply:
Yes